src/cli/commands/retrieve_history.go

```diff
@@ -55,6 +55,8 @@ func (command HistoryCommand) Execute([]string) error {
 		if rn <= 0 || err != nil {
 			return errors.New(fmt.Sprintf(ui.InvalidRecordNumber, command.RecordNumber))
 		}
+	} else if command.StartTime != "" || command.EndTime != "" {
+		rn = math.MaxInt64
```

Comment:
I would like to suggest to just do the strconv here, and  delegate the `if` statement in https://github.com/cloudfoundry-incubator/app-autoscaler-cli-plugin/blob/18c7a5fbd7e55f98c6df7bae0de8ace5d5af04cd/src/cli/commands/retrieve_history.go#L53-L63 to  https://github.com/cloudfoundry-incubator/app-autoscaler-cli-plugin/blob/18c7a5fbd7e55f98c6df7bae0de8ace5d5af04cd/src/cli/commands/retrieve_history.go#L134-L138 to make the logic more clearly.

Reply:
it's hard to move this part of the code, because i have to decide the `RecordNumber` from the initial `StartTime` and `EndTime` inputs